kong/clustering/services/sync/rpc.lua

```diff
@@ -407,12 +408,22 @@ function sync_once_impl(premature, retry_count)
 
   -- retry if the version is not updated
   retry_count = retry_count or 0
+  -- we do not count a timed out sync. just retry
+  if err ~= "timeout" then
+    retry_count = retry_count + 1
+  end
+
+  -- in some cases, the new spawned timer will be switched to immediately,
+  -- preventing the coroutine who possesses the mutex to run
+  -- to let other coroutines has a chance to run
+  yield()
```

Comment:
Could we just use a `0.001` delay in start_sync_once_timer()?

Reply:
could you elaborate?

Reply:
`kong.timer:at(0.01, sync_once_impl, retry_count or 0)`